migrations/2022-10-06-183632_move_blocklist_to_db/up.sql

```diff
@@ -0,0 +1,177 @@
+-- Create an instance table
+-- Holds any connected or unconnected domain
+create table instance (
+  id serial primary key,
+  domain varchar(255) not null unique,
+  published timestamp not null default now(),
+  updated timestamp null
+);
+
+-- Insert all the domains to the instance table
+insert into instance (domain)
+select distinct substring(p.actor_id from '(?:.*://)?(?:www\.)?([^/?]*)') from ( 
+  select actor_id from site 
+  union 
+  select actor_id from person 
+  union 
+  select actor_id from community
+) as p;
+
+-- Alter site, person, and community tables to reference the instance table.
+alter table site add column 
+instance_id int references instance on update cascade on delete cascade;
+
+alter table person add column 
+instance_id int references instance on update cascade on delete cascade;
+
+alter table community add column 
+instance_id int references instance on update cascade on delete cascade;
+
+-- Add those columns
+update site set instance_id = i.id 
+from instance i
+where substring(actor_id from '(?:.*://)?(?:www\.)?([^/?]*)') = i.domain;
+
+update person set instance_id = i.id 
+from instance i
+where substring(actor_id from '(?:.*://)?(?:www\.)?([^/?]*)') = i.domain;
+
+update community set instance_id = i.id 
+from instance i
+where substring(actor_id from '(?:.*://)?(?:www\.)?([^/?]*)') = i.domain;
+
+-- Make those columns unique not null now
+alter table site alter column instance_id set not null;
+alter table site add constraint idx_site_instance_unique unique (instance_id);
+
+alter table person alter column instance_id set not null;
+alter table community alter column instance_id set not null;
+
+-- Create allowlist and blocklist tables
+create table allowlist (
```

Comment:
Maybe call these like `federation_allowlist`, to make it clear what its about.

Reply:
Can do.